### ISSUE TITLE
Rearranging The website:

### ISSUE THREAD

**Issue by asterwolf** [AICSC/aicsc.github.io#78]
I'm adding this issue up as I'm currently interested in putting some work to rearrange things onto the website. The current home page is where it displays our meeting minutes; with this, I would like to make a separate page to display meeting minutes with a link on top next to constitution, while our home page represents more of an actual home page.

I see that default.html {{content}} is grabbing content from a file called index.html. Would this change be as simple as renaming index.html to something like minutes.html, create a new index.html and design that with the looks of a home page. Of course add the link for minutes on top and when clicked, it displays the instructions of minutes.html. 

I'm sure when making these changes, even if it is this simple, there will be simple things to change in other files as well. I wouldn't know what would need to be changed where to make sure things continue to run perfectly. I can for the most part trace everything that's happening pretty fine, my downfall is lack of knowledge of php and how it's working, and other minor things.

I'm thinking of just creating a new branch, copy the entire master branch over, and that will be our testing station for members who are interested to make changes onto. Once the site is exactly as we want it to be, we can just merge all changes to master. Of course that will be quite the project. 

Any advice?

**Comment by pathawks** [comment]
So, maybe it wasn't the most obvious thing to do, but the homepage is generated by [`util/index.html`](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/util/index.html). The reason that this becomes the homepage is all about the [`permalink: "/"`](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/util/index.html#L3) in the files front matter (the data in between the `---` and `---`).

If you wanted to change the URL of this page, you would just need to change the permalink to be something like `permalink: "/minutes/"` or whatever you wanted the new URL to be. (Maybe rename that source file too, just to avoid confusion.)

_Then_, to create a new homepage, you would just create a new file (either Markdown or HTML, your choice), and give it some front matter.

```yaml
---
permalink: "/"
layout: default
---
```

This front matter does two things: First, it says that the URL where this page will live is `/`, or at the root of the site.
Second, it says that after this page has been rendered, it should be dropped into [the `{{ content }}` section](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/_layouts/default.html#L27) of the layout [`_layouts/default.html`](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/_layouts/default.html).

Keep in mind that you can create as many new layouts as you need, and layouts can also have layouts.

In the layout [`_layouts/page.html`](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/_layouts/page.html), we create a title and a little block of content, and that's about it. However, once this is done, that is again dropped into the `{{ content }}` section of the layout `_layouts/default.html` because of the setting provided in [its front matter.](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/_layouts/page.html#L2)

The minutes you typically post don't have front matter. That's because I have defined their settings [inside the global `_config.yml`](https://github.com/AICSC/aicsc.github.io/blob/1fb21325fba1975c1ea0e69ba7621164d4d33318/_config.yml#L25-L29). (YAML is the data format for storing all that data, if you need to Google something.)

**Comment by pathawks** [comment]
I think it is a great idea for you to create new branches and try new things.

On the command line, if you want to create a new branch, you can just type:

```sh
git checkout -b MyCoolNewFeature
```

This will create a new branch called `MyCoolNewFeature` and immediately switch you to it, so that any new commits you create will be on that branch. Once you have it looking the way you want, you can

```
git push origin MyCoolNewFeature
```

This will push all of your committed changes up to GitHub, where you can create a pull request.

If you ever need to start over, you can

```sh
git reset --hard master
```

This will undo any commits that you have made on your branch, and set everything back to the way it is in the `master` branch. You can replace `master` with a commit ID, like 1fb21325fba1975c1ea0e69ba7621164d4d33318 to set everything back to the way it was as of that commit.

If you want to keep the files to retain your edits, but you just want to undo all of the commits on your branch, you can leave out the `--hard` and just

```sh
git reset master
```

Sometimes, when a pull request has too many commits, we will do something like this to squash several commits down to just a couple. This can make the history of a feature easier to understand for whichever poor sap in the future gets the job of debugging your old code 😛

**Comment by pathawks** [comment]
My favorite Git site: https://git-scm.com/

Jekyll documentation: https://jekyllrb.com/

Jekyll community help site: https://talk.jekyllrb.com/

If you ever post a question on [talk.jekyllrb.com](https://talk.jekyllrb.com/), please also @-mention me or maybe ping me on Slack and I will do what I can to help you out. :+1:

**Comment by asterwolf** [comment]
Right on, Thanks Pat! I have a good idea of where to begin now. Extremely helpful step by step. 👍

**Comment by asterwolf** [comment]
For now I decided to clone the repository to my desktop and mess around with it on my desktop for testing and learning purposes. Once I get a lot more familiar, I will make a new branch for others to watch and add their inputs.

**Comment by asterwolf** [comment]
In the file default.html line 18, it is controlling the adding of links to our nav bar  by grabbing the files from the folder pages. Of course, the order of pages on our nav bar are grabbed in alphabetical order, because the files are in alphabetical order, but if I wanted to decide the order, would changing the file names and adding maybe a number (and have them in chronological order in which I want the pages in the nav bar to be displayed), is that a decent approach? Or would that be bad practice? 

I hope I used proper terminology and you understand what I'm asking. 😛

**Comment by pathawks** [comment]
If you want more control over how those links are displayed, the best (and also easiest) thing to do might be to just replace that loop with the HTML of whatever links you need in whatever order you’d like them displayed.

Occam’s Razor of Programming :+1:

**Comment by asterwolf** [comment]
Yes, that could be done. But It's pretty convenient to add a new page in the page file and add a title to it. :p. Okay right on.

**Comment by pathawks** [comment]
> Yes, that could be done. But It's pretty convenient to add a new page in the page file and add a title to it.

In that case, sure, prepend a number to each filename :+1:
